build.sbt

```diff
@@ -91,19 +107,45 @@ lazy val samples = project
     }
   )
 
-lazy val tools = project in file("tools")
+lazy val tools = project("tools")
+
+lazy val sbtPlugin = project("sbt-scala-native-bindgen", ScriptedPlugin)
+  .dependsOn(tools)
+  .settings(
+    Keys.sbtPlugin := true,
+    scriptedLaunchOpts += s"-Dplugin.version=${version.value}",
+    scriptedLaunchOpts += {
+      val rootDir = (ThisBuild / baseDirectory).value
+      s"-Dbindgen.path=$rootDir/bindgen/target/scala-native-bindgen"
+    },
+    publishLocal := publishLocal.dependsOn(tools / publishLocal).value
+  )
 
-lazy val docs = project
-  .in(file("docs"))
+lazy val docs = project("docs")
   .enablePlugins(GhpagesPlugin, ParadoxSitePlugin, ParadoxMaterialThemePlugin)
   .settings(
     paradoxProperties in Paradox ++= Map(
       "github.base_url" -> scmInfo.value.get.browseUrl.toString
     ),
     ParadoxMaterialThemePlugin.paradoxMaterialThemeSettings(Paradox),
-    paradoxMaterialTheme in Paradox := {
-      (paradoxMaterialTheme in Paradox).value
+    Paradox / paradoxMaterialTheme := {
+      (Paradox / paradoxMaterialTheme).value
         .withRepository(scmInfo.value.get.browseUrl.toURI)
         .withColor("indigo", "indigo")
     }
   )
+
+def project(name: String, plugged: AutoPlugin*) = {
+  val unplugged = Seq(ScriptedPlugin).filterNot(plugged.toSet)
+  Project(id = name, base = file(name))
+    .disablePlugins(unplugged: _*)
+    .settings(
+      crossSbtVersions := List(Versions.sbt013, Versions.sbt1),
+      scalaVersion := {
+        (pluginCrossBuild / sbtBinaryVersion).value match {
+          case "0.13" => Versions.scala210
+          case _      => Versions.scala212
+        }
+      }
+    )
+}
```

Comment:
What was the problem with Scala/sbt versions? How does this `project` function solve it?

Reply:
The problem was that Scala Native requires `scalaVersion` 2.11 and sbt either 2.10 or 2.12. Except for the samples project which tests the  generated bindings, all other sbt projects use the `scalaVersion` associated with sbt.

Reply:
👍